### ISSUE TITLE
Reproducing the G-TAD results on THUMOS'14

### ISSUE THREAD

**Issue by zhang-can** [HumamAlwassel/TSP#9]
Hi @HumamAlwassel, thanks for your inspiring work!

Following [your reply](https://github.com/HumamAlwassel/TSP/issues/8#issuecomment-900516035), I interpolated the [features](https://github.com/HumamAlwassel/TSP#for-activitynet-v13-dataset) to [100, 400] and successfully reproduced the G-TAD results on ActivityNet v1.3. 

However, I only got 41.7% mAP@0.5 on THUMOS'14 dataset (paper: 43.2%). Here is what I did:
1. For each video features, linearly interpolate temporal dim according to the original TSN features used in GTAD.
2. For each video features, linearly interpolate channel dim to 2048 (1024 & 512 are also tried, but lower than 2048).
3. Uncomment all pre-processing code that loading flow features.

Is there any problem with the above implementation?
Thanks a lot!

**Comment by HumamAlwassel** [comment]
Hi @zhang-can ,

Thanks for your interest in our work.

I'm sorry but you may have misunderstood my response. The linear interpolation is only done on the temporal dimension and **_not_** on both the temporal and feature dimensions. For example in G-TAD for Activityvet, you should interpolate TSP features of size T x 512 to be 100 x 512. The interpolation is only done on the first dimension. Similarly, you should only interpolate the first dimension in G-TAD for THUMOS. 

Call the `torch.functional.interpolate()` similar the following for *each* video: 
```
from torch.functional import interpolate 

X = ... # numpy array from original G-TAD features of size [T1, F1] where T1 is the number of features in the video and F1 is the feature size
Y = ... # numpy array from TSP features of size [T2, F2] where F2 = 512 in this case

# transposed Y to be [1, F2, T2] so the interpolation is done on the correct dimension 
# the unsqueeze is needed because interpolate() expects 3D tensor
transposed_Y = Y.unsqueeze(0).permute(0,2,1)

interpolated_Y = interpolate(torch.tensor(transposed_Y), size=(X.shape[0]), mode='linear')) # shape [1, F2, T1]

# transpose back to the correct order
interpolated_Y = interpolated_Y.squeeze(0).permute(1,0).numpy() # shape [T1, F2]
```

Hope this helps. Cheers!
